.github/workflows/reusable_promote_prerelease.yaml

```diff
@@ -0,0 +1,60 @@
+name: Promote Prerelease to Release
+
+# This workflow checks for the latest successful prerelease, excluding any labeled with "artifacts-pending" or "pre-release-failure".
+# If a suitable prerelease is found, it promotes it to a full release. This action should consequently trigger any workflows
+# that respond to the 'released' event type, ultimately invoking 'reusable_on_release.yaml'. This facilitates an automated
+# continuation of the release process.
+
+#  Usage from the caller workflow:
+#  jobs:
+#    promote_prerelease:
+#      uses: newrelic/coreint-automation/.github/workflows/reusable_promote_prerelease.yaml@v2
+#      secrets:
+#        bot_token: ${{ secrets."github_token_name" }}
+
+on:
+  workflow_call:
+    inputs:
+      block_endpoint:
+        description: This endpoint should respond status 200 for the promotion to continue.
+        required: false
+        type: string
+        default: https://newrelic.github.io/coreint-automation/automatic_release_enable
+    secrets:
+      bot_token:
+        description: "github token"
+        required: true
+
+jobs:
+  check_block_endpoint:
+    name: Check block endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check block endpoint
+        run: |
+          http_status=$(curl -s -o /dev/null -w "%{http_code}" "${{ inputs.block_endpoint }}")
+
+          if [ "$http_status" != 200 ]; then
+            echo "Fail checking release block endpoint: ${{ inputs.block_endpoint }}"
+            exit 1
+          fi
+
+  promote_prerelease:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Promote previous prerelease to release
+        env:
+          GH_TOKEN: "${{ secrets.bot_token }}"
+        run: |
+          # Fetch the latest release information
+          read LATEST_RELEASE_TAG RELEASE_STATUS <<<$(gh release list --repo ${{ github.repository }} --limit 1 | awk '{print $1, $(NF-2)}')
+          RELEASE_STATUS_LOWER=$(echo "$RELEASE_STATUS" | tr '[:upper:]' '[:lower:]')
+
+          # Check if the latest release is a prerelease and not blocked
+          if [[ "$RELEASE_STATUS_LOWER" == "pre-release" ]]; then
+            echo "Latest release $LATEST_RELEASE_TAG is a prerelease and ready for promotion."
+            # Promote the prerelease to a full release
+            gh release edit "$LATEST_RELEASE_TAG" --repo ${{ github.repository }} --draft=false --prerelease=false
+          else
+            echo "The latest release $LATEST_RELEASE_TAG is not a prerelease, or it does not exist."
```

Comment:
Would it make sense to send a slack notification if the latest pre-release cannot be promoted because it has some status in the title? Eg: the pre-release still has `(artifacts-pending)` (after one week if everything was automatic!)

Reply:
Makes sense, going to add it 👍

Reply:
Done in 33709f6cbba4f498152de975ab81e7c964a96a63.